sdk/tables/Azure.Data.Tables/tests/TableClientQueryableLiveTests.cs

```diff
@@ -361,6 +399,108 @@ public async Task TableQueryableOnSupportedTypes()
             Assert.That(results.Count, Is.EqualTo(2));
         }
 
+        [Test]
+        public async Task TableQueryableWithDictionaryTypeOnSupportedTypes()
+        {
+            var entitiesToInsert = CreateComplexTableEntities(PartitionKeyValue, 4);
+            ComplexEntity thirdEntity = entitiesToInsert[2];
+
+            // Insert the new entities.
+
+            foreach (var entity in entitiesToInsert)
+            {
+                await client.InsertAsync(entity).ConfigureAwait(false);
+            }
+
+            // 1. Filter on String
+            var results = (await client.QueryAsync(ent => (ent["String"] as string).CompareTo(thirdEntity.String) >= 0).ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+            Assert.That(results.Count, Is.EqualTo(2));
+
+            // 2. Filter on Guid
+            results = (await client.QueryAsync(ent => (Guid)ent["Guid"] == thirdEntity.Guid).ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+            Assert.That(results.Count, Is.EqualTo(1));
+
+
+            // 3. Filter on Long
+            results = (await client.QueryAsync(ent => (Int64)ent["Int64"] >= thirdEntity.Int64).ToEnumerableAsync().ConfigureAwait(false)).ToList();
```

Comment:
Wonder if we should add an `ToListAsync` helper.

Reply:
Locally, or for `AsyncPageable` ?

Reply:
To test helpers in https://github.com/Azure/azure-sdk-for-net/blob/b9bd85cb2591f7c3f3c054f33f158e758056f1d3/sdk/core/Azure.Core.TestFramework/src/TestAsyncEnumerableExtensions.cs 

But now looking at the method it already returns the list.